toml/src/main/kotlin/org/rust/toml/inspections/CrateNotFoundInspection.kt

```diff
@@ -7,109 +7,26 @@ package org.rust.toml.inspections
+                    CratesLocalIndexService.getInstance().getCrate(dependency.crateName)
+                } catch (e: CratesLocalIndexException) {
+                    return
                 }
-            }
-
-            override fun visitTable(element: TomlTable) {
-                val depTable = DependencyTable.fromTomlTable(element) ?: return
-                if (depTable !is DependencyTable.Specific) return
-
-                handleDependency(
-                    DependencyCrate(depTable.crateName, depTable.crateNameElement, collectProperties(element)), holder
-                )
-            }
-        }
-    }
-
-    private fun handleDependency(dependency: DependencyCrate, holder: ProblemsHolder) {
-        // Do not check local, custom or git crates
-        for (property in IGNORED_PROPERTIES) {
-            if (property in dependency.properties) return
-        }
-
-        val crate = try {
-            CratesLocalIndexService.getInstance().getCrate(dependency.crateName)
-        } catch (e: CratesLocalIndexException) {
-            return
-        }
-
-        if (crate == null) {
-            holder.registerProblem(dependency.crateNameElement, "Crate ${dependency.crateName} not found")
-        }
-    }
-
-    companion object {
-        private val IGNORED_PROPERTIES = listOf("git", "path", "registry")
-    }
-}
-
-private fun collectProperties(owner: TomlKeyValueOwner): Map<String, TomlValue> {
-    return owner.entries.mapNotNull {
-        val name = it.key.name ?: return@mapNotNull null
-        val value = it.value ?: return@mapNotNull null
-        name to value
-    }.toMap()
-}
-
-private data class DependencyCrate(
-    val crateName: String,
-    val crateNameElement: TomlKeySegment,
-    val properties: Map<String, TomlValue>
-)
-
-private sealed class DependencyTable {
-    object General : DependencyTable()
-    data class Specific(val crateName: String, val crateNameElement: TomlKeySegment) : DependencyTable()
-
-    companion object {
-        fun fromTomlTable(table: TomlTable): DependencyTable? {
-            val key = table.header.key ?: return null
-            val segments = key.segments
-            val dependencyNameIndex = segments.indexOfFirst { it.isDependencyKey }
 
-            return when {
-                // [dependencies], [x86.dev-dependencies], etc.
-                dependencyNameIndex == segments.lastIndex -> General
-                // [dependencies.crate]
-                dependencyNameIndex != -1 -> {
-                    val crate = segments.getOrNull(dependencyNameIndex + 1)
-                    val crateName = crate?.name
-                    if (crate != null && crateName != null) {
-                        Specific(crateName, crate)
-                    } else {
-                        null
-                    }
+                if (crate == null) {
+                    holder.registerProblem(dependency.crateNameElement, "Crate ${dependency.crateName} not found")
```

Comment:
[minor] Let's introduce `val crateName = dependency.crateName` to replace all 3 `dependency.crateName` occurrences